traitsui/wx/scrubber_editor.py

```diff
@@ -346,6 +346,7 @@ def _pop_up_text(self):
         text.SetFocus()
         control.Bind(wx.EVT_TEXT_ENTER, self._text_completed, id=text.GetId())
         text.Bind(wx.EVT_KILL_FOCUS, self._text_completed)
+        text.Bind(wx.EVT_KILL_FOCUS, self._left_down)
```

Comment:
Thank you for all these efforts tracking down and fixing the segfault.

Could you please explain how this works (useful for future references too)?
For instance, the event emitted by `EVT_KILL_FOCUS` is an instance of [`wx.FocusEvent`](https://wxpython.org/Phoenix/docs/html/wx.FocusEvent.html#wx-focusevent).
I don't think the event supports the `GetX()` and `GetY()` methods needed by `self._left_down`, so I am quite confused...

Reply:
I think I might know why this seems to fix the issue...
It _seems_ the wx calls event handlers in the reverse order of when they are bound (I don't know if the order is documented or not).

For example, if I swap the order of `text.Bind(wx.EVT_KILL_FOCUS, self._text_completed)` and `text.Bind(wx.EVT_KILL_FOCUS, self._left_down)`, and I will get the segfault.

Here, the fact that `_left_down` causes an exception with `FocusEvent` prevented `self._text_completed` from being called, and the segfault seems to come from `self._text_completed`.

Reply:
With that understanding... adding `return` in `self._text_completed` after `event.Skip()` seems to achieve the same effect as the changes added here.
IOW, this:
```
    def _text_completed(self, event):
        """ Handles the user pressing the 'Enter' key in the text control.
        """
        if isinstance(event, wx.FocusEvent):
            event.Skip()
            return        # <---- here
        if self._update_value(event):
            self._destroy_text()
```

Reply:
In fact... that would be the same as removing `text.Bind(wx.EVT_KILL_FOCUS, self._text_completed)` altogether :)

While this results in a reduction in feature (the editor is probably meant to replicate the effect of hitting enter on the text box while it loses focus), the feature does not work and causes segfault. By removing it, we at least resolve the crippling segfault. The removed feature can be added back separately if we ever need it back.